techdocs/script/operations.py

```diff
@@ -116,11 +116,39 @@ def detect(self, filesystem):
                 self.index.remove(item)
 
         for item in child_result_files_to_be_copied:
-            self.index.add(IndexItem(item, self.repo))
+            self.index.add(FileIndexItem(item, self.repo))
         return child_result
 
 
-class Index:
+def hash(some_str):
+    return sha256(some_str.encode()).hexdigest()
+
+
+def hashb(data):
+    return sha256(data).hexdigest()
+
+
+class UnnecessaryOperationsFilteringDetector:
+    def __init__(self, child) -> None:
+        self.child = child
+
+    def detect(self, filesystem):
+        return list(filter(self.requires_changes_on_fs(filesystem), self.child.detect(filesystem)))
+
+    def requires_changes_on_fs(self, filesystem):
+        def inner(operation):
+            if operation.type == operation.TYPE_DELETE:
+                return True
+            if (not filesystem.is_file(operation.destination_abs)) or hashb(
+                filesystem.read_bytes(operation.source_abs)
+            ) != hashb(filesystem.read_bytes(operation.destination_abs)):
+                return True
+            return False
+
+        return inner
```

Comment:
I'm not seeing any test for it :see_no_evil: 

I assume that this local/remote branch comparison that we discuss will be a part of action itself right? https://piwikpro.atlassian.net/browse/ARCH-248

Reply:
Added